python/neuroglancer/pipeline/storage.py

```diff
@@ -204,6 +205,8 @@ class FileInterface(object):
 
     def __init__(self, path):
         self._path = path
+        self.dir_creation = Lock()
```

Comment:
Using locks might be a good idea versus recovering from errors, but testing demonstrates random deadlocks on `test/test_storage.py::test_list`

Another reason this may not be the right implementation is that each interface thread creating and acquiring a lock to work with means that the lock does not synchronize between threads. The threads must share a common lock for it to be meaningful.